File: Benchmarking/src/gen_113_16.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1): p0(v0), p1(v1) { }
  };
  struct s1 {
    s0 p0;
    std::vector<s0> p1;
    s1(s0 v0, std::vector<s0> v1): p0(v0), p1(v1) { }
  };
  struct s2 {
    std::vector<s0> p0;
    s1 p1;
    s2(std::vector<s0> v0, s1 v1): p0(v0), p1(v1) { }
  };
  __attribute__((noinline))
  double f0(const s2 &v0, const std::vector<s1> &v1, const double &v2) {
    const s1 v5 = v1[0];
    const std::vector<s0> v9 = v5.p1;
    const s0 v10 = v9[0];
    double v4 = v2;
    const std::vector<std::vector<double>> v11 = v10.p1;
    std::vector<std::vector<double>> v8 = v11;
    std::vector<std::vector<double>> v12 = v8;
    const std::vector<double> v13 = v12[0];
    const double v35 = v13[0];
    const std::vector<std::vector<double>> v34 = v10.p1;
    v8 = v34;
    const double v53 = v35 - v4;
    return v53;
  }
  int main() {
    s2 v0({ { { { { 0.0 }, { 1.0 } }, { { 2.0 } } }, { { { 3.0 }, { 4.0 } }, { { 5.0 } } } }, { { { { 6.0 }, { 7.0 } }, { { 8.0 } } }, { { { { 9.0 }, { 10.0 } }, { { 11.0 } } } } } });
    std::vector<s1> v1({ { { { { 12.0 }, { 13.0 } }, { { 14.0 } } }, { { { { 15.0 }, { 16.0 } }, { { 17.0 } } } } } });
    double v2(18.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }

File: Benchmarking/src/gen_113_16.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: s0
    var p1: [s0]
  }
  struct s2 {
    var p0: [s0]
    var p1: s1
  }
  @inline(never)
  func f0(_ v0: s2, _ v1: [s1], _ v2: Double) -> Double {
    let v5: s1 = v1[0]
    let v9: [s0] = v5.p1
    let v10: s0 = v9[0]
    var v4: Double = v2
    let v11: [[Double]] = v10.p1
    var v8: [[Double]] = v11
    var v12: [[Double]] = v8
    let v13: [Double] = v12[0]
    let v35: Double = v13[0]
    let v34: [[Double]] = v10.p1
    v8 = v34
    let v53: Double = v35 - v4
    return v53
  }
  func benchmark() {
    let v0: s2 = s2(p0: [s0(p0: [[0.0], [1.0]], p1: [[2.0]]), s0(p0: [[3.0], [4.0]], p1: [[5.0]])], p1: s1(p0: s0(p0: [[6.0], [7.0]], p1: [[8.0]]), p1: [s0(p0: [[9.0], [10.0]], p1: [[11.0]])]))
    let v1: [s1] = [s1(p0: s0(p0: [[12.0], [13.0]], p1: [[14.0]]), p1: [s0(p0: [[15.0], [16.0]], p1: [[17.0]])])]
    let v2: Double = 18.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 18.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()
